File: test/org/stringtemplate/v4/test/ErrorBufferAllErrors.java
package org.stringtemplate.v4.test;

import org.stringtemplate.v4.misc.ErrorBuffer;
import org.stringtemplate.v4.misc.STMessage;

public class ErrorBufferAllErrors extends ErrorBuffer {
	@Override
	public void runTimeError(STMessage msg) {
		errors.add(msg);
	}
}

File: objc/main/test/objc/ErrorBufferAllErrors.m
#import "ErrorBufferAllErrors.h"

@implementation ErrorBufferAllErrors

+ (id) newErrorBuffer
{
    return [[ErrorBufferAllErrors alloc] init];
}

- (id) init
{
    self = [super init];
    return self;
}

- (void) runTimeError:(STMessage *)msg {
  [errors addObject:msg];
}

@end
